src/metabase/models/card.clj

```diff
@@ -520,8 +537,10 @@
 
 (t2/define-after-select :model/Card
   [card]
-  (public-settings/remove-public-uuid-if-public-sharing-is-disabled
-   (dissoc card :dataset_query_metrics_v2_migration_backup)))
+  (-> card
+      (dissoc :dataset_query_metrics_v2_migration_backup)
+      (m/assoc-some :source_card_id (-> card :dataset_query source-card-id))
```

Comment:
Does this overwrite the `source_card_id` fetched from the db? Should it only be done when nil?